resources/views/actions/edit.blade.php

```diff
@@ -0,0 +1,100 @@
+                                value="{{ $action->name }}"
+                            @endif
+                        >
+                    </div>
+
+                    <div class="select form-item">
+                        <label class="field-label">Post Type</label>
+                        <select name="post_type">
+                            <option value="">Select the post type</option>
+                            @foreach($postTypes as $postType)
+                                @if (old('post_type'))
+                                    <option value="{{ $postType }}" {{ (old('post_type') == $postType ? "selected":"") }}>{{ $postType }}</option>
+                                @else
+                                    <option value="{{ $postType }}" {{ ($action->post_type == $postType ? "selected":"") }}>{{ $postType }}</option>
+                                @endif
+                            @endforeach
+                        </select>
+                    </div>
+                    <div class="form-item -third">
+                        <label class="field-label">CallPower Campaign ID</label>
+                        <input type="text" name="callpower_campaign_id" class="text-field" placeholder="e.g. 4 (optional)"
+                            @if (old('callpower_campaign_id'))
+                                value="{{ old('callpower_campaign_id') }}"
+                            @else
+                                value="{{ $action->callpower_campaign_id }}"
+                            @endif
+                        >
+                    </div>
+                    <div class="form-item -third">
+                        <label class="field-label">Action Noun</label>
+                        <input type="text" name="noun" class="text-field" placeholder="e.g. Jeans"
+                            @if (old('noun'))
+                                value="{{ old('noun') }}"
+                            @else
+                                value="{{ $action->noun }}"
+                            @endif
+                        >
+                    </div>
+                    <div class="form-item -third">
+                        <label class="field-label">Action Verb</label>
+                        <input type="text" name="verb" class="text-field" placeholder="e.g. Collected"
+                            @if (old('verb'))
+                                value="{{ old('verb') }}"
+                            @else
+                                value="{{ $action->verb }}"
+                            @endif
+                        >
+                    </div>
+                    <div class="form-item">
+                        <label class="field-label">Post Details</label>
+                        <label class="option -checkbox">
+                          <input type="checkbox" id=reportback name=reportback value=1 {{ old('reportback') || $action->reportback ? 'checked="checked"' : '' }}>
```

Comment:
This doesn't work perfectly right now. For example, if the Action is a reportback, and I want to edit it to not be a reportback but there is a validation error, I would expect the form to keep it unchecked when the validation error pops up with `old('reportback')` here. However, this changes the checkbox back to being checked when the validation error pops up because it satisfies `$action->reportback` since it hasn't been updated yet. 

I'm not sure how to get this to work the way I want it to because I can't really depend on discerning at what stage this is at (if it the second time and we want to grab `old('reportback')` vs `$action->reportback`). Kind of hard to explain via text so lmk if anyone wants to talk through this! (Also tried to do a Giphy Capture but for some reason wouldn't let me press "UPDATE ACTION" 🤔 ).

Reply:
I get what you're saying @chloealee ! Hmmm and if you remove the `$action->reportback` piece it won't be checked when you first get to the page to edit if it should be?

Reply:
@katiecrane yes exactly! It's so funky!

Reply:
@chloealee what is the value of `old('reportback')` if the user has not altered the form? Is it `false` or `null`? I'm wondering if there is a way to tell from that whether or not the user has touched that part of the form yet

Reply:
@katiecrane ooo that is a great call! From the [docs](https://laravel.com/docs/5.8/requests#retrieving-old-input), looks like it is `null`! However, I tried to do something like 
```
@if ($action->reportback)
  checked="checked"
@elseif (old('reportback')
  checked="checked"
@endif
```
but within an `@if` block, I don't have access to the `$action`'s reportback for some reason 🤔

Reply:
@katiecrane finally got this to work as expected! 
- On initial load, if the value is true for the action, the checkbox will be checked.
- If there are form validations, then the old values that the user wanted to uncheck/check are reflected, instead of what the database values.

Reply:
@chloealee YAYAYYYYY GREAT FIDDLING!!! 🎻